common/dev/pex89000.c

```diff
@@ -131,231 +118,375 @@ static bool pex89000_chime_read(uint8_t bus, uint8_t addr, uint32_t oft, uint8_t
+			rc = pex_api_unspecific_err;
+		}
+		break;
+
+	case pex_access_rev_id:
+		if (pex89000_chime_to_axi_read(bus, addr, BRCM_REG_CHIP_REVID, resp)) {
+			printf("%s: REVISION ID access failed!\n", __func__);
+			rc = pex_api_unspecific_err;
+		}
+		break;
+
+	case pex_access_sbr_ver:
+		if (pex89000_chime_to_axi_read(bus, addr, BRCM_REG_SBR_ID, resp)) {
+			printf("%s: SVR VERSION access failed!\n", __func__);
+			rc = pex_api_unspecific_err;
+		}
+		break;
+
+	case pex_access_flash_ver:
+		if (pex89000_chime_to_axi_read(bus, addr, BRCM_REG_FLASH_VER, resp)) {
+			printf("%s: FLASH VERSION access failed!\n", __func__);
+			rc = pex_api_unspecific_err;
+		}
+		break;
+
+	default:
+		printf("%s: Invalid key %d\n", __func__, key);
+		rc = pex_api_unspecific_err;
+		break;
+	}
+
+	if (k_mutex_unlock(&p->mutex))
+		printf("%s: pex89000 mutex %d unlock failed!\n", __func__, p->idx);
+
+	return rc;
+}
+
+static uint8_t pex89000_temp(uint8_t bus, uint8_t addr, pex_dev_t dev, uint32_t *val)
+{
+	uint8_t rc = pex_api_unspecific_err;
+
+	if (!val) {
+		printf("%s: *val does not exist !\n", __func__);
+		return rc;
+	}
+
+	float pre_highest_temp = 0;
+	float temp = 0;
+	float temp_arr[12];
+	uint32_t CmdAddr;
 	uint32_t resp = 0;
 
-	//check 0xFFE78504 value
-	if (!pex89000_chime_to_axi_read(bus, addr, BRCM_TEMP_SENSOR0_CTL_REG1, &resp)) {
-		if (resp != BRCM_TEMP_SENSOR0_CTL_REG1_RESET) {
+	if (dev == pex_dev_atlas1) {
+		if (pex89000_chime_to_axi_read(bus, addr, BRCM_REG_TEMP_SNR0_CTL, &resp)) {
+			printf("CHIME to AXI Read 0xFFE78504 fail!\n");
+			goto exit;
+		}
+		if (resp != BRCM_VAL_TEMP_SNR0_CTL_RESET) {
 			printf("ADC temperature control register1 check fail!\n");
-			return false;
+			goto exit;
+		}
+
+		if (pex89000_chime_to_axi_write(bus, addr, BRCM_REG_TEMP_SNR0_CTL,
+						BRCM_VAL_TEMP_SNR0_CTL_RESET)) {
+			printf("CHIME to AXI Write 0xFFE78504 fail!\n");
+			goto exit;
+		}
+
+		if (pex89000_chime_to_axi_read(bus, addr, BRCM_REG_TEMP_SNR0_STAT, &resp)) {
+			printf("CHIME to AXI Write 0xFFE78538 fail!\n");
+			goto exit;
+		}
+
+		temp = (resp & 0xFFFF) / 128;
+	} else if (dev == pex_dev_atlas2) {
+		for (int8_t i = 7; i < 12; i++) {
+			CmdAddr = (0x21 << 16) | (0x4C << 8) | (0x0B);
+			if (pex89000_chime_to_axi_write(bus, addr, 0xFFE00004, CmdAddr)) {
+				printf("CHIME to AXI Write 0xFFE00004 fail!\n");
+				goto exit;
+			}
+
+			if (pex89000_chime_to_axi_write(bus, addr, 0xFFE00008, i | 0x10000)) {
+				printf("CHIME to AXI Write 0xFFE00008 fail!\n");
+				goto exit;
+			}
+
+			CmdAddr = (0x22 << 16) | (0x4C << 8) | (0x14);
+			if (pex89000_chime_to_axi_write(bus, addr, 0xFFE0000C, CmdAddr)) {
+				printf("CHIME to AXI Write 0xFFE0000C fail!\n");
+				goto exit;
+			}
+
+			if (pex89000_chime_to_axi_read(bus, addr, 0xFFE00010, &resp)) {
+				printf("CHIME to AXI Write 0xFFE00010 fail!\n");
+				goto exit;
+			}
+
+			temp = (float)(366.812 - 0.23751 * (float)(resp & 0x7FF));
+			temp_arr[i] = temp;
```

Comment:
Can you please add comments describing what all these addresses are for?
Possibly add `#define` for the various constant values used here.

Reply:
Due to new PEX89000 chip is on development stage, some registers or data are not in current spec. Instead, those registers came from Broadcom debug tool. 
Dummy macro would be taken temporarily, thanks.